File: server/read/fastify/src/models/ArticleReadData.ts
import { Article } from '../entities/Article';

export class ArticleReadData {
  boardUUID: string;
  uuid: string;
  createdByUUID: string;
  title: string;
  titleHash: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;

  constructor(private article: Article) {
    this.uuid = article.uuid;
    this.title = article.title;
    this.titleHash = article.titleHash;
    this.content = article.content;
    this.createdAt = article.createdAt;
    this.updatedAt = article.updatedAt;
  }

  async lazyFill() {
    const board = await this.article.board;
    this.boardUUID = board.uuid;
    const createdBy = await this.article.createdBy;
    this.createdByUUID = createdBy.uuid;
  }
}

File: server/write/spring/src/main/kotlin/com/board/server/write/spring/models/ArticleReadData.kt
package com.board.server.write.spring.models

import com.board.server.write.spring.entities.ArticleEntity
import java.time.LocalDateTime
import java.util.*

data class ArticleReadData(
    private val article: ArticleEntity,
    val boardUUID: UUID,
    val createdByUUID: UUID,
) {
    val uuid: UUID = article.uuid
    val title: String = article.title
    val titleHash: String = article.titleHash
    val content: String = article.content
    val createdAt: LocalDateTime = article.createdAt
    val updatedAt: LocalDateTime = article.updatedAt
}
